posts/2021/04/rethinking-jupyter-documentation.md

```diff
@@ -0,0 +1,192 @@
+which attempts to auto reformat and fix common NumyDoc's format mistakes and
+typos – but that's a subject of a future post.
+
+# Stuck between a Rock and a Hard place
+
+While Sphinx and related project are great at offering hosted HTML
+documentation; extensive usage of those make interactive documentation harder to
+consume;
+
+While it is possible to [run Sphinx on the fly when rendering
+docstrings](https://github.com/spyder-ide/docrepr), most Sphinx features
+only work when building a full project, with the proper configuration and
+extension and can be computationally intensive. This make running Sphinx locally
+impractical.
+
+Hosted website often may not reflect locally installed version of the
+libraries and requires careful linking, deprecation and narrative around
+platform or version specific features.
+
+# This is fixable
+
+For the past few month I've been working on rewriting how IPython (and hence
+Jupyter) can display documentation. It works both in terminal (IPython) and
+browser context (notebook, JupyterLab, Spyder) with proper rendering, and currently
+understand most directives; it could be customized to understand any new ones:
+
+<img alt="papyri1" src="/images/2021/04/papyri-1.png" class='center' >
+
+Above is the (terminal) documentation of `scipy.polynomial.lagfit`, see how the
+single backticks are properly understood and refer to known parameters, it
+detected that  `` `n` `` is incorrect as it should have double backticks; notice
+the rendering of the math even in terminal.
+
+For that matter technically this does not care as to whether the DocString is
+written in RST or Markdown; though I need to implement the later part. I believe
+though that some maintainers would be quite happy to use markdown whose syntax
+more users are familiar with.
+
+<img alt="papyri navigation" src="/images/2021/04/papyri-nav.gif" class='center' >
+
+It support navigation – here in terminal – where clicking or pressing enter on a
+link would bring you to the target page. In above gif you can see that many
+token of code example are also automatically type-inferred (thanks [Jedi](https://github.com/davidhalter/jedi)), and
+can also be clicked to navigate to their corresponding page.
+
+<img alt="papyri terminal-fig" src="/images/2021/04/papyri-terminal-fig.png" class='center' >
+
+Images are included, even in terminal when they are not inline but replaced by
+a button to open them in your preferred viewer (see the `Open with quicklook` in
+above screenshot).
+
+# The future
+
+
+I'm working on a number of other features, in particular :
+
+ - rendering of narrative docs – for which I have a prototype,
+ - automatic indexing of all the figures and plots –  working but slow right now.
+ - proper cross library reference and indexing without the need for intersphinx.
+
+    - It is possible from the `numpy.linspace` page to see all page that
+      reference it, or use it in an example (see previous image).
+
```

Comment:
The second indentation looks a bit strange [when rendered](https://deploy-preview-184--quansight-labs.netlify.app/blog/2021/04/rethinking-jupyter-documentation/). I would suggest either moving this out one level or concatenating it to the previous paragrahp